server/src/main/java/org/apache/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -282,6 +282,13 @@ public void stop()
   public void add(String lookupName, LookupExtractorFactoryContainer lookupExtractorFactoryContainer)
   {
     Preconditions.checkState(lifecycleLock.awaitStarted(1, TimeUnit.MILLISECONDS));
+    LookupLoadingSpec lookupLoadingSpec = lookupListeningAnnouncerConfig.getLookupLoadingSpec();
+    if (lookupLoadingSpec.getMode() == LookupLoadingSpec.Mode.NONE ||
+        (lookupLoadingSpec.getMode() == LookupLoadingSpec.Mode.ONLY_REQUIRED
+         && !lookupLoadingSpec.getLookupsToLoad().contains(lookupName))) {
+      LOG.warn("Skipping notice to add lookup [%s] : current lookup loading mode does not allow it.", lookupName);
+      return;
```

Comment:
I think the `SeekableStreamIndexTask` should populate the task [context](https://github.com/apache/druid/blob/master/indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTask.java#L103) from the context map similar to how a few other task types do. Then the change in the lookup manager may not be necessary since the lookup spec would already have been materialized